lib/hosts.nix

```
@@ -3,7 +3,7 @@
   nixpkgs,
   ...
 }: rec {
-  mkHost = base: system: extraModules:
+  mkHost = base: system: extraModules: hostname:
```

```suggestion
  mkHost = base: system: hostname: extraModules:
```
Would be more concise.

Even better solution since this is getting big is probably to turn this into an attribute set with defaults, i.e.:
```suggestion
  mkHost = { base, hostname, system ? "x86_64-linux", extraModules ? [] }:
```

---

So, the thing is that I quite prefer the currying that is going on with the function arguments way.
It allows you to define more systems easily as clones of existing ones with e.g. in steward.nix: `prod = mkSteward x86_64-linux [...]`, and then `prod1 = prod "prod1.attest.profian.com"; prod2 = prod ....;`.

Regarding hostnames etc: I have another commit inbound for setting the hostname based on a regex split version of the passed in name here, but I was trying to keep the different commits separate.

---

The `imports = ... ${config.networking.fqdn}` is a good one that I'll add in that commit.

---

Thinking about that further, adding a `domain:` argument would allow that even better, because you could even set the domain already for the `prod = ` argument, so I think I'll do that, and update this PR.